lib/rubocop/haml/ruby_clipper.rb

```diff
@@ -96,6 +96,7 @@ class TrailingDoRemover
           [ \t]*
           (\|[^|]*\|)?
           [ \t]*
+          (\#.*)?
```

Comment:
Thanks for the pull request. I hadn't considered the comment at the end, but you are right.

```suggestion
          (?=[ \t]*(?:\#.*)?\Z)
```

What about using `(?=...)`? This way, I think we can remove only the ` do` part and leave the `# ...` part.

Reply:
I had a similar idea but it keeping code comment while stripping code _in between_ won't work well with the offset and RuboCop will report wrong position for code comments which contain, say, unnecessary RuboCop directives.

Ideally, we would not strip any `do |x|` blocks but add an `end` instead but that change would be more involved I guess :shrug: